### PR TITLE
Add Timer Datatype

### PR DESCRIPTION
**Pull request by Headline**
Timers don't have very many functions attributed to the Handle, so this is quite a small methodmap. This is one step towards methodmap completion, though.

This pull request allows for sourcemod plugin creators to make timers like so

```cpp
Timer timer = new Timer(10.0, Timer_Callback);
...
timer.Kill();
```

### CODE REVIEW

**Comment by powerlord** [comment]
The only catch I can see is that it will break any plugins that pass timer callbacks around.

**Comment by Headline22** [comment]
This is true, but I do not believe any plugins do this.

File: plugins/include/timers.inc [hunk 1]
```diff
@@ -45,7 +45,7 @@
 /**
  * Any of the following prototypes will work for a timed function.
  */
-typeset Timer
```
Comment (peace-maker) [inline plugins/include/timers.inc]: You should change the tag of the `Handle timer` in the callback signatures to `Timer timer` too.

Comment (Headline) [reply plugins/include/timers.inc]: Absolutely right. Thanks pm

File: plugins/include/timers.inc [hunk 2]
```diff
@@ -201,9 +233,9 @@ native bool IsServerProcessing();
  * @param flags				Timer flags.
  * @return					Handle to the timer object.  You do not need to call CloseHandle().
  */
-stock Handle CreateDataTimer(float interval, Timer func, Handle &datapack, int flags=0)
+stock Timer CreateDataTimer(float interval, TimerCallback func, Handle &datapack, int flags=0)
```
Comment (ZeronTheXeon) [inline plugins/include/timers.inc]: Handle &datapack -> DataPack &datapack

Comment (Headline) [reply plugins/include/timers.inc]: 👍

**Comment by Deathreus** [comment]
While you're refactoring the timer include:
```/**
 * Returns whether or not the server is processing frames or not.  
 *
 * The server does not process frames until at least one client joins the game.  
 * Once the first player has in, even if that player, leaves, the server's 
 * timers and entities will work.
 *
 * @return					True if the server is ticking, false otherwise.
 */
native bool IsServerProcessing();
```
That brief bugs my inner grammar nazi

**Comment by JoyKoy** [comment]
Is this gonna be merged?

**Comment by KyleSanderson** [comment]
We can't land this, it's going to break close to a decades worth of plugins.

**Comment by asherkin** [comment]
That is no reason to close it - it can just remain open until the next worthwhile widespread breaking change.

**Comment by Headline** [comment]
@asherkin i'll remember this so when that time comes we can bring this back

**Pull request closed without merging**